client/src/pages/record.tsx

```diff
@@ -219,36 +226,47 @@ function RecordPage(props: {
       });
   }
 
+  function onTrimVideo(startTime: number, endTime: number) {
+    if (!mentor || !curAnswer) {
+      return;
+    }
+    setLoadingMessage("Trimming video...");
+    trimVideo(mentor._id, curAnswer.question._id, startTime, endTime)
+      .then((job) => {
+        setTrimStatusUrl(job.statusUrl);
+        setIsTrimming(true);
+      })
+      .catch((err) => {
+        toast(`Trim failed`);
+        console.error(err);
+        setLoadingMessage(undefined);
+        setIsTrimming(false);
+      });
+  }
+
+  function onRerecordVideo() {
+    if (!curAnswer) {
+      return;
+    }
+    setCurAnswer({ ...curAnswer, recordedAt: "" });
+  }
+
   useInterval(
     (isCancelled) => {
-      fetchUploadVideoStatus(statusUrl)
+      fetchUploadVideoStatus(uploadStatusUrl)
```

Comment:
for the isCancelled below, we only have cancelled as true if this page is unmounted, so don't set any state in that case, just return